src/index.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable semi */
+/* eslint-disable no-param-reassign */
+/* eslint-disable arrow-body-style */
+/* eslint-disable no-restricted-globals */
+/* eslint-disable object-curly-newline */
+/* eslint-disable object-curly-spacing */
+/* eslint-disable camelcase */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-undef */
+/* eslint-disable no-return-assign */
+/* eslint-disable arrow-parens */
+const weatherKey = '6080e17b0fc07b99a71fde330eb48aed';
+const searchForm = document.querySelector('.check-by-city .search-form');
+const searchCity = document.querySelector('.check-by-city .search-city');
+const warning = document.querySelector('.check-by-city .warning');
+const iconImg = document.querySelector('.icon-container > img');
+const desc = document.querySelector('.icon-container > p');
+const cityTemp = document.querySelector('.city-temp > div');
+const cityCountry = document.querySelectorAll('.city-country > span');
+const extraInfo = document.querySelectorAll('.extra-info > p');
+const a = document.getElementById('switchKel');
+const b = document.getElementById('temp-toggle');
+const c = document.getElementById('main-temp');
+const d = document.getElementById('btn-main-temp');
+
+/* eslint no-return-assign: ["error", "always"] */
+const getWholeData = (data) => {
+  // eslint-disable-next-line arrow-body-style
+  const getWeather = () => {
+    return {description, icon, main} = data.weather[0];
+  };
+  const getTemp = () => {
+    return {temp, temp_max, temp_min, feels_like} = data.main;
+  };
+  // eslint-disable-next-line arrow-body-style
+  const getPlace = () => {
+    // eslint-disable-next-line no-global-assign
+    return [name, country] = [data.name, data.sys.country];
+  };
+  return { getWeather, getTemp, getPlace };
+};
+const dataArray = [];
+searchForm.addEventListener('submit', event => {
```

Comment:
This is a huge block of code inside the event listener, you could try to refactor/extract some pieces of it into reusable functions, that's going to make it more readable/maintainable/debuggable.